scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1062,6 +1086,28 @@
                               :first-ten-jobs (print-str (->> jobs (take 10) (map :job/uuid) (map str)))}))
       (dissoc compute-cluster->jobs :no-acceptable-compute-cluster))))
 
+(defn distribute-and-launch-jobs
+  "Assign each job to a compute cluster (using configuration such as pool name and cluster
+   capability) and launch them in parallel.
+
+   A function `launch-distributed-job-fn` is used to actually do the launching. This is 
+   needed to implement the different launch logic for launching synthetic pods for 
+   autoscaling or for using the Kubernetes Scheduler directly."
+  [jobs pool-name compute-clusters job->acceptable-compute-clusters-fn launch-distributed-job-fn]
+  (try
+    (when (seq jobs)
+      (log-structured/info "Preparing for job distribution" {:pool pool-name})
+      (let [compute-cluster->jobs (distribute-jobs-to-compute-clusters jobs pool-name compute-clusters
+                                                                       job->acceptable-compute-clusters-fn)]
+        (log-structured/info "Starting job distribution" {:pool pool-name})
+        (->> compute-cluster->jobs
+             (map (fn [launch-args] (future (launch-distributed-job-fn launch-args))))
```

Comment:
This can be an anonyomous function with the syntax:   `#(future (launch-distributed-jobs-fn %))`